faunadb/serialization_test.go

```diff
@@ -1819,3 +1819,81 @@ func TestSerializeAccessProviders(t *testing.T) {
 		`{"access_providers":{"database":"auth_db"}}`,
 	)
 }
+
+func TestSerializeStructWithOmitEmptyTags(t *testing.T) {
+	type TestStruct struct{}
+	type OmitStruct struct {
+		Name           string      `fauna:"name,omitempty"`
+		Age            int         `fauna:"age,omitempty"`
+		Payment        float64     `fauna:"payment,omitempty"`
+		AgePointer     *int        `fauna:"agePointer,omitempty"`
+		PaymentPointer *float64    `fauna:"paymentPointer,omitempty"`
+		Struct         *TestStruct `fauna:"struct,omitempty"`
+	}
+	x := 10
+	y := 42.42
+	z := TestStruct{}
+	tests := []struct {
+		name string
+		expr Expr
+		want string
+	}{
+		{
+			name: "Empty Int",
+			expr: Obj{"data": OmitStruct{Name: "John", Age: 0}},
```

Comment:
Is this what other libraries do, treat 0 as "empty"? This is very weird behavior because 0 is a valid int value

Reply:
it will treat this value as empty only when field declared as `omitempty`.

Reply:
I see this, what I'm questioning is treat 0 as empty given 0 is a valid value, this will not play well if you have a multi driver application. Given the Go driver will omit the 0 fields, other drivers will have to treat this special case where the non existent field should be treated as 0, also the other drivers don't do this treatment, 0 will be persisted as is.

Reply:
There is some sort of convention related to `omitempty`:
_The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string._
So users should wight the pros and cons when using it.

Also there is a workaround - users can specify field as pointer and omitempty:
```go
type OmitStruct struct {
	Name           string      `fauna:"name,omitempty"`
	AgePointer     *int        `fauna:"agePointer,omitempty"`
}
p := 0
{
	name: "Pointer int",
	expr: Obj{"data": OmitStruct{Name: "John", AgePointer: &p}},
	want: `{"object":{"data":{"object":{"name":"John", "agePointer":0}}}}`,
},
```